src/components/RecipeEdit.vue

```diff
@@ -129,6 +141,19 @@ export default {
         addEntry: function(field, index, content='') {
             this.recipe[field].splice(index, 0, content)
         },
+        beforeWindowUnload(e) {
+          if (this.confirmStayInEditedForm()) {
+            // Cancel the window unload event
+            e.preventDefault()
+            e.returnValue = ''
+          }   
+        },
+        confirmLeavingPage() {
+          return window.confirm('You have unsaved changes! Do you still want to leave?')
+        },
+        confirmStayInEditedForm() {
+          return !this.savingRecipe && this.formDirty && !this.confirmLeavingPage()
```

Comment:
This will continue to new URL when a saving action is just running, right?

Reply:
The idea was: If the user clicked the save button (saving is running), the browser continues to the new URL without checking if the form is dirty (`this.formDirty`) and (if it is dirty) asking for confirmation (`!this.confirmLeavingPage()`)

Reply:
Yes, I see that. This is more related to the overall comment below about the abort of a saving action.